test/unittest/tests/block/test_block_session.cpp

```diff
@@ -0,0 +1,505 @@
+    {
+        WT_EXT *ext;
+        WT_EXT *cached_ext;
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+        // Construct extent cache with one item.
+        bms->ext_cache = ext;
+        bms->ext_cache_cnt = 1;
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &cached_ext) == 0);
+        REQUIRE(cached_ext == ext);
+        validate_and_cleanup_ext_block(ext);
+    }
+
+    SECTION("Allocate with two extents in cache ")
+    {
+        WT_EXT *ext;
+        WT_EXT *ext2;
+        WT_EXT *cached_ext;
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext2) == 0);
+
+        // Construct extent cache with two items.
+        ext->next[0] = ext2;
+        bms->ext_cache = ext;
+        bms->ext_cache_cnt = 2;
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &cached_ext) == 0);
+        REQUIRE(ext == cached_ext);
+        REQUIRE(ext2 != cached_ext);
+        validate_and_cleanup_ext_block(ext);
+        validate_and_cleanup_ext_block(ext2);
+    }
+}
+
+TEST_CASE("Block session: __wti_block_ext_free", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    SECTION("Free with null block manager session -- needs discussion")
+    {
+        std::shared_ptr<MockSession> session_no_bm = MockSession::buildTestMockSession();
+        WT_EXT *ext;
+
+        REQUIRE(__ut_block_ext_alloc(session_no_bm->getWtSessionImpl(), &ext) == 0);
+        REQUIRE(ext != nullptr);
+
+        __wti_block_ext_free(session_no_bm->getWtSessionImpl(), ext);
+        // REQUIRE(ext == nullptr);
+    }
+
+    SECTION("Calling free with cache")
+    {
+        WT_EXT *ext;
+        WT_EXT *ext2;
+
+        REQUIRE(__ut_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+
+        __wti_block_ext_free(session->getWtSessionImpl(), ext);
+
+        REQUIRE(ext != nullptr);
+        REQUIRE(bms->ext_cache == ext);
+
+        REQUIRE(__ut_block_ext_alloc(session->getWtSessionImpl(), &ext2) == 0);
+        __wti_block_ext_free(session->getWtSessionImpl(), ext2);
+
+        REQUIRE(ext != nullptr);
+        REQUIRE(bms->ext_cache == ext2);
+        REQUIRE(bms->ext_cache->next[0] == ext);
+        validate_and_cleanup_ext_list(bms, 2);
+    }
+}
+
+TEST_CASE("Block session: __wti_block_ext_prealloc", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+
+    SECTION("Pre-alloc with null block manager")
+    {
+        WT_BLOCK_MGR_SESSION *bms;
+
+        bms = nullptr;
+        __wt_random_init(&session->getWtSessionImpl()->rnd);
+
+        REQUIRE(__wti_block_ext_prealloc(session->getWtSessionImpl(), 0) == 0);
+        bms = static_cast<WT_BLOCK_MGR_SESSION *>(session->getWtSessionImpl()->block_manager);
+
```

Comment:
We could also check  session->block_manager_cleanup = __block_manager_session_cleanup here. It's also worth commenting why this is the case.

Reply:
I could only check if it is not null, the __block_manager_session_cleanup function is internal to the file. We can't expose it as well because that would create a different address for the function e.g. __ut_block_manager_session_cleanup has different address to __block_manager_session_cleanup

Reply:
That's true, then verifying it was NULL and now isn't NULL would be valuable.